src/mlpack/methods/neighbor_search/ns_model.hpp

```diff
@@ -37,6 +49,127 @@ struct NSModelName<FurthestNeighborSort>
   static const std::string Name() { return "furthest_neighbor_search_model"; }
 };
 
+class MonoSearchVisitor : public boost::static_visitor<void>
```

Comment:
Actually, I should add, if you don't want to or are busy doing other things, I can do the documentation here, it will only take a few minutes.